app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -81,7 +81,9 @@ def timestamp
       timestamp_param = params.permit(:timestamp)[:timestamp]
       return nil if timestamp_param.nil?
 
-      Time.strptime(timestamp_param, '%Y-%m-%dT%H:%M:%S%z')
+      date_fmt = timestamp_param.match?('\.') ? '%Y-%m-%dT%H:%M:%S.%N%z' : '%Y-%m-%dT%H:%M:%S%z'
```

Comment:
I'd do a literal string check instead of a regex, less worrying about escaping.

```suggestion
      date_fmt = timestamp_param.include?('.') ? '%Y-%m-%dT%H:%M:%S.%N%z' : '%Y-%m-%dT%H:%M:%S%z'
```